File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mxprofit=0
        minstake=prices[0]
        for i in range(1,len(prices)):
            if prices[i]<minstake:
                minstake=prices[i]
            else:
                if prices[i]-minstake>mxprofit:
                    mxprofit=prices[i]-minstake
        return mxprofit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
import java.util.List;

class Solution {
    public int maxProfit(int[] prices) {
        int mxprofit = 0;
        int minstake = prices[0];
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < minstake) {
                minstake = prices[i];
            } else if (prices[i] - minstake > mxprofit) {
                mxprofit = prices[i] - minstake;
            }
        }
        return mxprofit;
    }
}